Sources/Controllers/DashboardOnMap/MapSettings/OAMapSettingsRoutesScreen.mm

```diff
@@ -0,0 +1,400 @@
+{
+    self = [super init];
+    if (self)
+    {
+        _app = [OsmAndApp instance];
+        _styleSettings = [OAMapStyleSettings sharedInstance];
+        _routesParameter = [_styleSettings getParameter:param];
+
+        if ([param isEqualToString:SHOW_CYCLE_ROUTES_ATTR])
+        {
+            _routesSettingType = ERoutesSettingCycle;
+            settingsScreen = EMapSettingsScreenCycleRoutes;
+            _routesEnabled = _routesParameter.storedValue.length > 0 && [_routesParameter.storedValue isEqualToString:@"true"];
+        }
+        else if ([param isEqualToString:HIKING_ROUTES_OSMC_ATTR])
+        {
+            _routesSettingType = ERoutesSettingHiking;
+            settingsScreen = EMapSettingsScreenHikingRoutes;
+            _routesEnabled = _routesParameter.storedValue.length > 0 && ![_routesParameter.storedValue isEqualToString:@"disabled"];
+        }
+        else
+        {
+            _routesSettingType = ERoutesSettingTravel;
+            settingsScreen = EMapSettingsScreenTravelRoutes;
+            _routesEnabled = _routesParameter.storedValue.length > 0;
+        }
+
+        vwController = viewController;
+        tblView = tableView;
+        _mapViewController = [OARootViewController instance].mapPanel.mapViewController;
+        [self commonInit];
+        [self initData];
+    }
+    return self;
+}
+
+- (void)commonInit
+{
+}
+
+- (void)initData
+{
+    NSMutableArray *dataArr = [@[
+            @[
+                    @{@"type": [OADividerCell getCellIdentifier]},
+                    @{@"type": [OASettingSwitchCell getCellIdentifier]},
+                    @{@"type": [OADividerCell getCellIdentifier]}
+            ]
+    ] mutableCopy];
+
+    NSMutableArray *colorsArr = [@[@{@"type": [OADividerCell getCellIdentifier]}] mutableCopy];
+    if (_routesSettingType == ERoutesSettingCycle)
+    {
+        [colorsArr addObject:@{
+                @"type": [OASettingsTitleTableViewCell getCellIdentifier],
+                @"value": @"false",
+                @"title": OALocalizedString(@"gpx_route")
+        }];
+        [colorsArr addObject:@{
+                @"type": [OASettingsTitleTableViewCell getCellIdentifier],
+                @"value": @"true",
+                @"title": OALocalizedString(@"rendering_value_walkingRoutesOSMCNodes_name")
+        }];
+    }
+    else
+    {
+        for (OAMapStyleParameterValue *value in _routesParameter.possibleValuesUnsorted)
+        {
+            if (value.name.length != 0)
+            {
+                [colorsArr addObject:@{
+                        @"type": [OASettingsTitleTableViewCell getCellIdentifier],
+                        @"value": value.name,
+                        @"title": value.title
+                }];
+            }
+        }
+    }
+    [colorsArr addObject:@{@"type": [OADividerCell getCellIdentifier]}];
+    [dataArr addObject:colorsArr];
+
+    _data = [NSArray arrayWithArray:dataArr];
```

Comment:
You can set NSMutableArray to an array here. It is completely safe as you lose the reference to the mutable array anyway.